instrumentation/log4j/log4j-2.13.2/library/src/main/java/io/opentelemetry/instrumentation/log4j/v2_13_2/LogEventMapper.java

```diff
@@ -0,0 +1,141 @@
+    levelSeverityMap.put(Level.DEBUG, Severity.DEBUG);
+    levelSeverityMap.put(Level.INFO, Severity.INFO);
+    levelSeverityMap.put(Level.WARN, Severity.WARN);
+    levelSeverityMap.put(Level.ERROR, Severity.ERROR);
+    levelSeverityMap.put(Level.FATAL, Severity.FATAL);
+    LEVEL_SEVERITY_MAP = Collections.unmodifiableMap(levelSeverityMap);
+  }
+
+  static void mapLogEvent(LogBuilder builder, LogEvent logEvent) {
+    AttributesBuilder attributes = Attributes.builder();
+
+    // message
+    Message message = logEvent.getMessage();
+    if (message != null) {
+      builder.setBody(message.getFormattedMessage());
+    }
+
+    // time
+    builder.setEpoch(logEvent.getNanoTime(), TimeUnit.NANOSECONDS);
+
+    // level
+    Level level = logEvent.getLevel();
+    if (level != null) {
+      builder.setSeverity(
+          LEVEL_SEVERITY_MAP.getOrDefault(level, Severity.UNDEFINED_SEVERITY_NUMBER));
+      builder.setSeverityText(logEvent.getLevel().name());
+    }
+
+    // logger
+    attributes.put(ATTR_LOGGER_NAME, logEvent.getLoggerName());
+
+    // fully qualified class name
+    attributes.put(ATTR_FQCN, logEvent.getLoggerFqcn());
+
+    // thread
+    attributes.put(ATTR_THREAD_NAME, logEvent.getThreadName());
+    attributes.put(ATTR_THREAD_ID, logEvent.getThreadId());
+    attributes.put(ATTR_THREAD_PRIORITY, logEvent.getThreadPriority());
+
+    // throwable
+    Throwable throwable = logEvent.getThrown();
+    if (throwable != null) {
+      attributes.put(ATTR_THROWABLE_NAME, throwable.getClass().getName());
+      attributes.put(ATTR_THROWABLE_MESSAGE, throwable.getMessage());
+    }
+
+    // marker
+    Marker marker = logEvent.getMarker();
+    if (marker != null) {
+      attributes.put(ATTR_MARKER, marker.getName());
+    }
+
+    // nested diagnostic context
+    List<String> contextStackList = logEvent.getContextStack().asList();
+    if (contextStackList.size() > 0) {
+      attributes.put(ATTR_NDC, contextStackList);
+    }
+
+    // mapped diagnostic context (included span context)
+    ReadOnlyStringMap contextData = logEvent.getContextData();
+    if (contextData != null) {
+      Map<String, String> contextMap = contextData.toMap();
+      if (contextMap != null) {
+        String traceId =
+            contextMap.containsKey(TRACE_ID) ? contextMap.remove(TRACE_ID) : TraceId.getInvalid();
+        String spanId =
+            contextMap.containsKey(SPAN_ID) ? contextMap.remove(SPAN_ID) : SpanId.getInvalid();
+        TraceFlags traceFlags =
+            contextMap.containsKey(TRACE_FLAGS)
+                ? TraceFlags.fromHex(contextMap.remove(TRACE_FLAGS), 0)
+                : TraceFlags.getDefault();
+        Context context =
+            Context.root()
+                .with(
+                    Span.wrap(
+                        SpanContext.create(traceId, spanId, traceFlags, TraceState.getDefault())));
+        builder.setContext(context);
+
+        contextMap.forEach(attributes::put);
```

Comment:
I don't think we should fill arbitrary attributes, I don't think we ever do anything similar for spans

Reply:
Kind of related to the the conversation about [doing a complete mapping](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/4375#discussion_r749527440) or not.

Not filling arbitrary attributes from mapped diagnostic context is limiting the usefulness of the log4j2 api. It means if you want to use otel logs in a more "machine generated events" type of way, that you'd want to use the otel log sdk in your application instead of the log4j2 api.

Reply:
I think we'd generally use baggage for more arbitrary attributes? Though we haven't even specced it for spans. But I'd still suggest keeping things smaller for now to make sure the spec grows rather than going all in